lib/puppet/type/http.rb

```diff
@@ -1,17 +1,79 @@
 Puppet::Type.newtype :http, :is_capability => true do
-  newparam(:name, :is_namevar => true) do
-    desc 'The name of the resource'
+  @doc = "A type for testing http connections"
+
+  ensurable do
+    defaultvalues
+    defaultto :present
   end
 
-  newparam(:ip) do
-    desc 'IP address of the HTTP service'
+  newparam(:name, :is_namevar => true) do
+    desc 'The unique name of the resource'
   end
 
   newparam(:port) do
-    desc 'Port of the HTTP service'
+    desc 'The TCP port that the HTTP service is listening to'
+    defaultto 80
+
+    validate do |value|
+      Integer(value)
+    end
+
+    munge do |value|
+      Integer(value)
+    end
+  end
+
+  newparam(:ip) do
+    desc 'The external IP address of the HTTP service'
   end
 
   newparam(:host) do
-    desc 'Hostname of the HTTP service'
+    desc 'The hostname of the HTTP service'
+  end
+
+  newparam(:ssl) do
+    desc 'Set to true if the HTTP service uses SSL encryption'
+    defaultto false
+  end
+
+  newparam(:path) do
+    desc 'Path of HTTP resource'
+    defaultto "/"
+  end
+
+  newparam(:timeout) do
+    desc 'Time before timing out the resource (seconds)'
+    defaultto 60
+
+    validate do |value|
+      Float(value)
+    end
+
+    munge do |value|
+      Float(value)
+    end
+  end
+
+  newparam(:ping_interval) do
+    desc 'Time to sleep before attempting to try a connection again (second)'
+    defaultto 1
+
+    validate do |value|
+      Float(value)
+    end
+
+    munge do |value|
```

Comment:
only accepting a number. I asked the module team. We can fix it before release if they're concerned.